x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_factory.ts

```diff
@@ -0,0 +1,324 @@
+            lastLookbackDate: runResult.lastLookbackDate,
+            searchAfterTimes: result.searchAfterTimes.concat(runResult.searchAfterTimes),
+            state: runState,
+            success: result.success && runResult.success,
+            warning: warningMessages.length > 0,
+            warningMessages,
+          };
+          runState = runResult.state;
+        }
+
+        if (result.warningMessages.length) {
+          const warningMessage = buildRuleMessage(result.warningMessages.join());
+          await ruleStatusService.partialFailure(warningMessage);
+        }
+
+        if (result.success) {
+          const createdSignalsCount = result.createdSignals.length;
+
+          if (actions.length) {
+            const notificationRuleParams: NotificationRuleTypeParams = ({
+              ...params,
+              name: name as string,
+              id: ruleSO.id as string,
+            } as unknown) as NotificationRuleTypeParams;
+
+            const fromInMs = parseScheduleDates(`now-${interval}`)?.format('x');
+            const toInMs = parseScheduleDates('now')?.format('x');
+            const resultsLink = getNotificationResultsLink({
+              from: fromInMs,
+              to: toInMs,
+              id: ruleSO.id,
+              kibanaSiemAppUrl: (meta as { kibana_siem_app_url?: string } | undefined)
+                ?.kibana_siem_app_url,
+            });
+
+            logger.info(buildRuleMessage(`Found ${createdSignalsCount} signals for notification.`));
+
+            if (createdSignalsCount) {
+              const alertInstance = services.alertInstanceFactory(alertId);
+              scheduleNotificationActions({
+                alertInstance,
+                signalsCount: createdSignalsCount,
+                signals: result.createdSignals,
+                resultsLink,
+                ruleParams: notificationRuleParams,
+              });
+            }
+          }
+
+          logger.debug(buildRuleMessage('[+] Signal Rule execution completed.'));
+          logger.debug(
+            buildRuleMessage(
+              `[+] Finished indexing ${createdSignalsCount} signals into alias ${indexAlias}`
+            )
+          );
+
+          if (!hasError && !wroteWarningStatus && !result.warning) {
+            await ruleStatusService.success('succeeded', {
+              bulkCreateTimeDurations: result.bulkCreateTimes,
+              searchAfterTimeDurations: result.searchAfterTimes,
+              lastLookBackDate: result.lastLookbackDate?.toISOString(),
+            });
+          }
+
+          // adding this log line so we can get some information from cloud
+          logger.info(
+            buildRuleMessage(
+              `[+] Finished indexing ${createdSignalsCount} ${
+                !isEmpty(tuples)
+                  ? `signals searched between date ranges ${JSON.stringify(tuples, null, 2)}`
+                  : ''
+              }`
+            )
+          );
+        } else {
```

Comment:
looks like the block starting with `if (result.success)` is a pretty large one, followed by two error handling scenarios, one for !result.success and one for caught error. I was wondering if it would be slightly more readable if we had one handleSuccess function and one handleError function, where the handleError one can be called in both places. handleSuccess one wouldn't exactly be reused, but it could be good for readability